pwdtk/auth_backends.py

```diff
@@ -14,19 +14,39 @@
 logger = logging.getLogger(__name__)
 
 
-class PwdtkLockedException(Exception):
-    """ custom exception """
+class PwdtkBaseException(Exception):
 
     def __init__(self, pwdtk_data):
 
         self.pwdtk_data = pwdtk_data
 
 
+class PwdtkLockedException(PwdtkBaseException):
+    """ custom exception """
+
+
+class PwdtkForceRenewException(PwdtkBaseException):
+    """ custom exception """
+
+
 class PwdtkBackend(ModelBackend):
     """
     Authenticates against settings.AUTH_USER_MODEL.
     """
+    def compute_failed_login(self, pwdtk_data):
```

Comment:
doc strings missing.
mention perhaps et least what data type pwdtk_data is

Reply:
done